### PR TITLE
Add `personal_sign` and `personal_ecRecover` RPC methods

### LINKED ISSUE #4460: Add `personal_sign` and `personal_recover`
Details:
https://github.com/ethereum/go-ethereum/pull/2940

### PR DESCRIPTION
**Pull request by andresilva**
Fixes #4460.

### CODE REVIEW

File: rpc/src/v1/impls/personal.rs [hunk 1]
```diff
@@ -133,6 +174,14 @@ impl<D: Dispatcher + 'static> Personal for PersonalClient<D> {
 		}
 	}
 
+	fn sign(&self, data: RpcBytes, account: RpcH160, password: String) -> BoxFuture<RpcH520> {
+		self.do_sign(data, account, password)
+	}
+
+	fn ec_recover(&self, data: RpcBytes, signature: RpcH520) -> BoxFuture<RpcH160> {
+		self.do_ec_recover(data, signature)
```
Comment (tomusdrw) [inline rpc/src/v1/impls/personal.rs]: Why is it not implemented inline?

**Comment by andresilva** [comment]
I tried to use cryptokitties with this and it made a call to `personal_sign` with only two arguments (missing password). According to https://github.com/ethereum/go-ethereum/wiki/Management-APIs#personal_sign I think the password should be mandatory, but it shows the password as optional for the console API. Should I implement the same behavior, i.e. if no password is provided the account should already be unlocked (same as `eth_sign`).

File: rpc/src/v1/impls/personal.rs (hunk 1 above)
Comment (andresilva) [reply rpc/src/v1/impls/personal.rs]: Inlined both.

**Comment by andresilva** [comment]
I've confirmed that geth requires the 3 parameters to be present using the RPC API, so we should leave it as is. I'm not sure why metamask has implemented it with only 2 arguments though.

**Comment by andresilva** [comment]
Can I rebase this to avoid the merges?

**Comment by 5chdn** [comment]
Yes, rebasing should work.

**Comment by 5chdn** [comment]
Please resolve conflicts :)

**Comment by andresilva** [comment]
Rebased.

**Comment by wliment** [comment]
@andresilva  i build from master branch source but `personal_ecRecover` method not found

```
curl --data '{"method":"personal_ecRecover","params":[ "Hello","0x190dc01f46fe8ad8ab57eeca71999329773d34d9eff000b91f6e4f947722868718e0bb85d396ea09f066ba0a46d4917ffa26f421833116abc4b62c8e91c5cd2e1b"],"id":1,"jsonrpc":"2.0"}' -H "Content-Type: application/json" -X POST localhost:8545
{"jsonrpc":"2.0","error":{"code":-32601,"message":"Method not found"},"id":1}
```

**Comment by 5chdn** [comment]
@wliment did you enable the personal jsonrpc module?

**Pull request merged**